clap_derive/src/derives/into_app.rs

```diff
@@ -296,7 +319,7 @@ pub fn gen_app_augmentation(
                     },
 
                     Ty::Other => {
-                        let required = !attrs.has_method("default_value");
+                        let required = !attrs.has_method("default_value") && !override_required;
```

Comment:
Why was this extra condition needed here? Is there a test case that tests the need for this?

Reply:
Partial updates need to relax the requirements. There is already a testcase for it:

c44cf07e78b4e82c792d50f6a8ade7025a040cda